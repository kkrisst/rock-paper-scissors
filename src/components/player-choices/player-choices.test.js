import React from 'react'
import { shallow } from 'enzyme'

import PlayerChoices from './player-choices.component.jsx'

describe('Testing the PlayerChoices component', () => {

  let handleChoiceMock = jest.fn();
  const wrapper = shallow(<PlayerChoices handleChoice={handleChoiceMock} />);

  it('expect PlayerChoices to be defined', () => {
    expect(PlayerChoices).toBeDefined();
  });

  it('expect PlayerChoices to match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('expect PlayerChoices to have choices', () => {
    const gameModeButtons = wrapper.find('.choice');
    expect(gameModeButtons.length).toEqual(3);
  });

  it('expect no choice to be selected', () => {
    const gameModeButtons = wrapper.find('.selected');
    expect(gameModeButtons.length).toEqual(0);
  });

  it('expect one choice to be selected', () => {
    const wrapper2 = shallow(<PlayerChoices selected={'rock'} />);
    const gameModeButtons = wrapper2.find('.selected');
    expect(gameModeButtons.length).toEqual(1);
  });

  it('expect handleChoice to be called on choice click with the correct parameter', () => {
    wrapper.find('.choice').at(0).simulate('click');
    expect(handleChoiceMock).toBeCalled();
    expect(handleChoiceMock).toBeCalledWith('rock');

    wrapper.find('.choice').at(1).simulate('click');
    expect(handleChoiceMock).toBeCalledWith('paper');

    wrapper.find('.choice').at(2).simulate('click');
    expect(handleChoiceMock).toBeCalledWith('scissors');
  });

})