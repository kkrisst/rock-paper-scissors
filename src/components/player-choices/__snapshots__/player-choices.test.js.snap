// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the PlayerChoices component expect PlayerChoices to match snapshot 1`] = `ShallowWrapper {}`;
